default/AI/AIstate.py

```diff
@@ -120,8 +127,28 @@ def __setstate__(self, state_dict):
         for odict_attrib in ['colonisablePlanetIDs', 'colonisableOutpostIDs']:
             if dict_attrib not in state_dict:
                 self.__dict__[odict_attrib] = odict()
+        if 'uuid' not in state_dict:
+            self.uuid = uuid4()
+        if 'turn_hashes' not in state_dict:
```

Comment:
Boy, I really don't like how they do the code excerpts here.  In the above one, I had selected lines 132 and 133 for comment, but it chooses to show lines 127-132 but not 133-- doesn't make much sense to me.  I mention this because the excerpt makes it look like I was talking about lines 130-132, and it doesn't matter if one of those talks about uuid and the other hashes.  But I wanted to make it more clear that my comment was that it seems to me that line 133 should match line 132.  Perhaps you already understood that's what I meant, but your response was so laconic it wasn't clear to me.